pkg/apis/rke.cattle.io/v1/plan/plan.go

```diff
@@ -1,13 +1,15 @@
 package plan
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	capi "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
 type Plan struct {
-	Nodes    map[string]*Node         `json:"nodes,omitempty"`
-	Machines map[string]*capi.Machine `json:"machines,omitempty"`
-	Cluster  *capi.Cluster            `json:"cluster,omitempty"`
+	Nodes    map[string]*Node          `json:"nodes,omitempty"`
+	Machines map[string]*capi.Machine  `json:"machines,omitempty"`
+	Secrets  map[string]*corev1.Secret `json:"secrets,omitempty"`
```

Comment:
After the secret is stored here, we really only care about the labels and annotations on it for the planner. The plan data is in the `Node` struct.

Reply:
If that's the case, then should we just store the `labels` and `annotations` in a map of secret names to a struct containing them then? I'm not sure we need to be dragging around all of the secret data around as it can get quite large